src/JiemyuDLL/Dll/Entities/Behaviors/Move/StandardMove.cs

```diff
@@ -0,0 +1,196 @@
+﻿using JiemyuDll.Entities.Behaviors;
+using JiemyuDll.Map;
+using Microsoft.Xna.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace JiemyuDll.Entities.Behaviors.Move
+{
+    class StandardMove : Move
+    {
+        private struct AStarValues
+        {
+            public int G;
+            public int H;
+            public int F;
+            public Vector2 Parent;
+        };
+
+        public uint Distance { get; set; }
+
+        // A dictionary of the cached paths for this move. That way we don't
+        // need to recalculate every time. A path is an ordered list of points
+        // for where to move. 
+        public Dictionary<Vector2, List<Vector2>> Paths = new Dictionary<Vector2, List<Vector2>>(); 
+ 
+        public override bool InMove(Move otherMove)
+        {
+            if (otherMove is StandardMove)
+            {
+                StandardMove stdMove = (StandardMove)otherMove;
```

Comment:
Maybe abstract Move should have distance?  I think it makes sense for any type of move to have a distance.  That way you don't have to cast here.  I'm not 100% on what this is being used for currently, but maybe it's even worth having InMove be in Move if it's just a distance calculation.